### ISSUE TITLE
Stack master: does not build haddocks with --no-haddock-deps after successful build

### ISSUE THREAD

**Issue by snoyberg** [commercialhaskell/stack#4754]
This is a regression on Stack master, tested with `ddca714c73701fcefd0d4493364f6661a543e73f`.

With Stack 1.9.3:

```
stack-1.9.3 new foo --resolver ghc-8.6.4 && cd foo && stack-1.9.3 build && stack-1.9.3 haddock --no-haddock-deps --open
```

results in the Haddocks for the library being generated. However, for Stack master:

```
stack new foo --resolver ghc-8.6.4 && cd foo && stack build && stack haddock --no-haddock-deps --open
```

results in the Haddocks not being generated. This is likely a relatively simple dirtiness checking bug, maybe related to configure caches, but I'm not certain.

Marking as P1, but not investigating myself immediately. The codepaths may change significantly for component-based builds, so I'd rather see that land first.

**Comment by qrilka** [comment]
It is a dirtiness check bug and `--no-haddock-deps` doesn't play any role here, I remember somewhat how I deleted haddocks check but it looks like it wasn't a correct decision.

**Comment by snoyberg** [comment]
Addressed by #4813, closing.
